### PR TITLE
Towards replacing uncrustify with clang-format in Axom

### PR DESCRIPTION
**Pull request by kennyweiss**
# Summary

- This PR adds infrastructure for replacing `uncrustify` with `clang-format` as Axom's formatter, as discussed at our most recent team meeting (and several times before that).
- I have tried to match axom's current style as much as possible. 
  - I started with `serac`'s clang-format file and added/modified options until the output looked reasonably close to axom.
  - There are some differences related to whitespaces where `uncrustify` was not consistent. (We probably set some of these options to `ignore`)
- I have not run `clang-format` on this branch. I will post that in a separate PR (see #292 ).
  - I manually edited a few files to fix issues that I discovered before running clang-format.
  - Hopefully, this will allow others on the team to play with some settings and suggest improvements.
  - We can also use this branch to mark regions of the code that should not be formatted with `"clang-format off"` and `"clang-format on"`
  - I manually edited the toss3 gcc@8 host-config to use the `clang-format from clang release 10. Once we've agreed on this, I'll update the remaining host-configs (and spack/uberenv).

TODO:
- [x] Update docs to reflect change from uncrustify to clang-format
- [x] Update `release-notes`
- [x] Remove `uncrustify` config file

Future PRs:
* [#335] Update spack/uberenv/tools to depend on clang-format instead of uncrustify
* [#336] Add `make check` to at least one CI job

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -51,7 +51,7 @@ if (“${PROJECT_SOURCE_DIR}” STREQUAL “${CMAKE_SOURCE_DIR}”)
     endif()
 
     set(ENABLE_ASTYLE      OFF CACHE BOOL "")
-    set(ENABLE_CLANGFORMAT OFF CACHE BOOL "")
+    set(ENABLE_UNCRUSFITY  OFF CACHE BOOL "")
```
Comment (kennyweiss) [inline src/CMakeLists.txt:54]:
```suggestion
    set(ENABLE_UNCRUSTIFY  OFF CACHE BOOL "")
```

File: src/axom/core/execution/internal/cuda_exec.hpp [hunk 2]
```diff
@@ -13,11 +13,11 @@
 #include "umpire/Umpire.hpp"
 
 #ifndef RAJA_ENABLE_CUDA
-#error *** CUDA_EXEC requires a CUDA enabled RAJA ***
+#error "*** CUDA_EXEC requires a CUDA enabled RAJA ***"
```
Comment (gzagaris) [inline src/axom/core/execution/internal/cuda_exec.hpp:16]: Here and in similar places, the quotes are not needed for the argument to the `#error` directive. I think this would actually print the error message with the `"`, which is not the intent.  Please, revert this change.

Comment (kennyweiss) [reply src/axom/core/execution/internal/cuda_exec.hpp:16]: Thanks @gzagaris.  I added them because clang-format removed the spaces between the `#error` and `***` and it became
```
 #error*** CUDA_EXEC requires a CUDA enabled RAJA *** 
```

What do you suggest?
* Adding quotes ?
* Removing the askerisks ?
* \<other> ?

File: src/axom/quest/tests/quest_all_nearest_neighbors.cpp [hunk 3]
```diff
@@ -168,7 +168,7 @@ TEST(quest_all_nearnbr, cplx_13region_query)
 {
   SLIC_INFO("*** 13-region closely-packed query.");
 
-  // Pre-formatted lines, not to be disturbed by uncrustify: *INDENT-OFF*
+  // Pre-formatted lines, not to be disturbed by clang format: clang-format off
```
Comment (white238) [inline src/axom/quest/tests/quest_all_nearest_neighbors.cpp:171]: I say just go to the normal comment and be consistent with the rest of the code.

**Comment by kennyweiss** [comment]
Does anyone have an estimate about when they'll have a chance to review / experiment with this PR?

File: src/axom/core/execution/internal/cuda_exec.hpp (hunk 2 above)
Comment (rhornung67) [reply src/axom/core/execution/internal/cuda_exec.hpp:16]: I would remove the asterisks and leave out the quotes. If it is hit, it will still print out the message indicating an error.

File: src/CMakeLists.txt (hunk 1 above)
Comment (kennyweiss) [reply src/CMakeLists.txt:54]: Good question @agcapps  -- `blt` will check for an `uncrustify` executable unless you explicitly diasble it with `ENABLE_UNCRUSTIFY=OFF`.  This line is explicitly disabling uncrustify since axom is no longer using it.

File: src/axom/core/execution/internal/cuda_exec.hpp (hunk 2 above)
Comment (kennyweiss) [reply src/axom/core/execution/internal/cuda_exec.hpp:16]: Done.  I removed the quotes and the asterisks,

File: src/axom/quest/tests/quest_all_nearest_neighbors.cpp (hunk 3 above)
Comment (kennyweiss) [reply src/axom/quest/tests/quest_all_nearest_neighbors.cpp:171]: Good suggestion.  I removed the stuff before `clang-format off`, as suggested.

**Comment by kennyweiss** [comment]
This is practically ready to merge.
* I've updated the documentation and RELEASE-NOTES in this PR and manually edited the host-configs to point to clang-format instead of uncrustify.
* I've applied the new formatting in #292, and am running all the LC tests now.

Next steps:
* I'll take care of updating the `dev-tools` and automatically updating the `host-configs` in a separate PR.
* Similarly, we'll have to add an automatic CI job to run `make check` in a separate PR.

I'll create issues for both of these.
